calculator_test.go

```diff
@@ -2,23 +2,227 @@ package calculator_test
 
 import (
 	"calculator"
+	"math"
+	"math/rand"
 	"testing"
+	"time"
 )
 
 func TestAdd(t *testing.T) {
 	t.Parallel()
-	var want float64 = 4
-	got := calculator.Add(2, 2)
-	if want != got {
-		t.Errorf("want %f, got %f", want, got)
+	type testCase struct {
```

Comment:
In practice, we probably wouldn't bother to give this type a name; we would make it an _anonymous_ struct type:

```go
testCases := []struct{
		name string
		a, b float64
		want float64
	}{
		{name: "Two plus two equals four", a: 2, b: 2, want: 4},
		...
	}
```

The syntax looks a little weird, but all we're really doing is defining the type 'on the fly', where it's needed.